src/seccomp/src/syscall_table.rs

```diff
@@ -0,0 +1,673 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+/// Creates and owns a mapping from the arch-specific syscall name to the right number.
+#[derive(Debug)]
+pub(crate) struct SyscallTable {
+    map: HashMap<String, i64>,
+    arch: String,
+}
+
+impl SyscallTable {
+    pub fn new(arch: String) -> Self {
+        let mut instance = Self {
+            arch,
+            map: HashMap::with_capacity(332), // nr of syscalls for x86_64 (upper bound)
+        };
+
+        instance.populate_map();
+
+        instance
+    }
+
+    /// Returns the arch-specific syscall number based on the given name.
+    /// Adheres to the "SYS_*" notation.
+    pub fn get_syscall_nr(&self, sys_name: &str) -> Option<i64> {
+        self.map.get(sys_name).copied()
+    }
+
+    // We need to create an architecture-specific map since the set of
+    // possible syscalls is architecture-dependent. We can't just map to their
```

Comment:
I agree that my comment is a bit misleading. I meant that we can't use the libc constants because at compile time, we can only get access to one target. This is because libc uses conditional compilation, while seccompiler needs access to the constants for all libc targets

Reply:
Also, rust libc seems to have the complete syscall table for x86_6-musl and aarch64-musl targets. Their list has been updated and I can't find any syscall that it lacks.

Reply:
That's great, and maybe we can use it :smile: in any case, I think your explanation about the compile-time restriction would make the comment less misleading.

Reply:
removed this comment. doesn't make that much sense now that we autogenerate the syscall tables